### PR TITLE
[develop] Remove special-casing of lookup-related columns

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR is #13547 against develop, using the proof-systems version in https://github.com/o1-labs/proof-systems/pull/1147. This PR builds upon https://github.com/MinaProtocol/mina/pull/13686.

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

File: src/lib/pickles_types/plonk_types.ml [hunk 1]
```diff
@@ -849,23 +794,26 @@ module Evals = struct
...
       (module Impl : Snarky_backendless.Snark_intf.Run with type field = f)
-      ~dummy e feature_flags :
+      ~dummy e (feature_flags : _ Features.t) :
       ((a_var, Impl.Boolean.var) In_circuit.t, a t, f) Snarky_backendless.Typ.t
       =
     let open Impl in
-    let lookup_typ = Lookup.opt_typ Impl.Boolean.typ feature_flags e ~dummy in
     let opt flag = Opt.typ Impl.Boolean.typ flag e ~dummy in
+    let lookup_sorted =
+      match feature_flags.lookup (* Fixme *) with
```
Comment (rbonichon) [inline src/lib/pickles_types/plonk_types.ml:811]: That's fixed in #13688.

File: src/lib/pickles_types/plonk_types.ml [hunk 2]
```diff
@@ -883,7 +831,10 @@ module Evals = struct
       ; opt feature_flags.foreign_field_mul
       ; opt feature_flags.xor
       ; opt feature_flags.rot
-      ; lookup_typ
+      ; opt feature_flags.lookup (* TODO: Fixme *)
+      ; opt feature_flags.lookup (* TODO: Fixme *)
+      ; Typ.array ~length:5 (opt lookup_sorted) (* TODO: Fixme *)
```
Comment (rbonichon) [inline src/lib/pickles_types/plonk_types.ml:836]: That's fixed in #13688. Good.

**Review by rbonichon — APPROVED** [review]
LGTM once you consider the fixes in #13688

**Comment by rbonichon** [comment]
The failure of `versions-compatibility-linter` is not important.

**Pull request merged**